public/bit_depth.h

```diff
@@ -12,113 +12,35 @@
-  //   7 bit weights, 5 bit activations, switch at 128:  59.58% top-1 accuracy
-  //   7 bit weights, 5 bit activations, switch at 192:  63.37% top-1 accuracy
-  //   7 bit weights, 5 bit activations, switch at 256:  63.47% top-1 accuracy
-  //   7 bit weights, 5 bit activations, switch at 320:  63.71% top-1 accuracy
-  //   7 bit weights, 5 bit activations, switch at 384:  63.71% top-1 accuracy
-  //   7 bit weights, 5 bit activations, switch at 448:  63.58% top-1 accuracy
-  //   7 bit weights, 5 bit activations, switch at 512:  64.10% top-1 accuracy
-  //   7 bit weights, 5 bit activations, switch at 640:  62.49% top-1 accuracy
-  //   7 bit weights, 5 bit activations, switch at 768:  62.49% top-1 accuracy
-  //   7 bit weights, 5 bit activations, switch at 1024: 58.96% top-1 accuracy
-  //
-  // So here, 384 looks comfortably in the middle of a plateau of good values,
-  // and it's a roundish number (3/2 * 256) so let's stick with that for now.
-  // It would be nice to work out the theory of this, and understand how this
-  // should depend on the distribution of inputs and the bit depth.
-  //
-  // Repeating the same evaluation with AddMod:
-  //   7 bit weights, 5 bit activations, switch at 64:   62.65% top-1 accuracy
-  //   7 bit weights, 5 bit activations, switch at 128:  62.65% top-1 accuracy
-  //   7 bit weights, 5 bit activations, switch at 192:  63.81% top-1 accuracy
-  //   7 bit weights, 5 bit activations, switch at 256:  64.23% top-1 accuracy
-  //   7 bit weights, 5 bit activations, switch at 320:  64.16% top-1 accuracy
-  //   7 bit weights, 5 bit activations, switch at 384:  64.16% top-1 accuracy
-  //   7 bit weights, 5 bit activations, switch at 448:  64.16% top-1 accuracy
-  //   7 bit weights, 5 bit activations, switch at 512:  64.52% top-1 accuracy
-  //   7 bit weights, 5 bit activations, switch at 640:  62.74% top-1 accuracy
-  //   7 bit weights, 5 bit activations, switch at 768:  62.74% top-1 accuracy
-  //   7 bit weights, 5 bit activations, switch at 1024: 59.74% top-1 accuracy
-  //
-  // The behavior is similar, so 384 remains a good choice.
-
-  static const int kRoundingModeSizeThreshold = 384;
-};
-
+// Default: LHS and RHS are 8bit.
 struct DefaultL8R8BitDepthParams {
   typedef BitDepth<8> LhsBitDepth;
   typedef BitDepth<8> RhsBitDepth;
-  typedef ExactRoundingStrategyFor8Bit RoundingStrategy;
 };
 
-struct DefaultL7R5BitDepthParams {
-  typedef BitDepth<7> LhsBitDepth;
-  typedef BitDepth<5> RhsBitDepth;
-  typedef DefaultRoundingStrategyForLessThan8Bit RoundingStrategy;
-};
+// Deprecated: when gemmlowp used to allow requantizing 8bit
+// inputs to less-than-8-bit depths, the public setting allowing
+// that was DefaultL7R5BitDepthParams. That requantization
+// feature has been removed, but as the whole point of that
+// requantization was to make less-than-8-bit an internal
+// optimization without any impact on the API (other than lowering
```

Comment:
Is there any easy way to log a warning if anyone's still executing this code path? I don't see anything internal, but it might be nice to give a clue to anyone outside who's not paying attention to the release notes. Just a nice to have though, don't worry if it's tricky to implement.